src/auth/auth.proto

```diff
@@ -190,92 +177,54 @@ message Groups {
 //// Authorization API
 
 message AuthorizeRequest {
-  // repo is the object that the caller wants to access
-  string repo = 1;
+  Resource resource = 1;
 
-  // scope is the access level that the caller needs to perform an action
-  Scope scope = 2;
+  // permissions are the operations the caller is attempting to perform 
+  repeated Permission permissions = 2;
 }
 
 message AuthorizeResponse {
-  // authorized is true if the caller has at least
-  // 'AuthorizeRequest.scope'-level access to 'AuthorizeRequest.repo', and false
-  // otherwise
+  // authorized is true if the caller has the require permissions
   bool authorized = 1;
-}
 
-message GetScopeRequest {
-  string username = 1;
+  // satisfied is the set of permission that the principal has
+  repeated Permission satisfied = 2;
 
-  // repos are the objects to which 'username's access level is being queried
-  repeated string repos = 2;
-}
+  // missing is the set of permissions that the principal lacks
+  repeated Permission missing = 3;
 
-message GetScopeResponse {
-  // scopes (actually a "role"--see "Scope") are the access level that
-  // 'GetScopeRequest.username' has to each repo in 'GetScopeRequest.repos', in
-  // the same order that repos appeared in 'repos'.
-  repeated Scope scopes = 1;
+  // principal is the principal the request was evaluated for
+  string principal = 4;
 }
 
-message SetScopeRequest {
-  // username is the principal (some of which belong to robots rather than
-  // users, but the name is preserved for now to provide compatibility with the
-  // pachyderm dash) whose access is being granted/revoked. As with
-  // GetScopeRequest, to set the access level of a robot user, the caller must
-  // prefix username with "robot:". 
-  string username = 1;
-
-  // repo is the object to which access is being granted/revoked
-  string repo = 2;
-
-  // scope (actually a "role"--see "Scope") is the access level that the owner
-  // of 'principal' will now have
-  Scope scope = 3;
-}
+message ModifyRoleBindingRequest {
+  // resource is the resource to modify the role bindings on 
+  Resource resource = 1;
 
-message SetScopeResponse {}
+  // principal is the principal to modify the roles binding for
+  string principal = 2;
 
-message GetACLRequest {
-  string repo = 1;
+  // roles is the set of roles for principal 
```

Comment:
Just checking my understanding: to revoke someone's access to a resource, you'd leave `roles` empty?

Reply:
Correct, I'll add a comment to that effect.